### PR TITLE
Carry segments over during complex memory access expressions

### PR DESCRIPTION
**Pull request by DerelictDrone**
Solves #10

Could use a more thorough look by someone with more time but as far as my tests indicate, this solves the issue with no side effects

Original Code:
image

Before fix:
279844664-b150f229-5139-4baf-b82d-46044c83725a

After fix:
image

Extra example:
image
image

### CODE REVIEW

**Review by thegrb93 — APPROVED** [review]

**Comment by DerelictDrone** [comment]
So, @thegrb93 will these get merged after a waiting period or does a second maintainer (like Vurv) need to review? Maybe a few people test the changes?

**Comment by thegrb93** [comment]
I'll merge after a little while if there's no new activity

**Pull request merged**
